operators/test/e2e/failure_test.go

```diff
@@ -99,18 +104,137 @@ func TestKillOneMasterNode(t *testing.T) {
+	sc.ObjectMeta = metav1.ObjectMeta{
+		Name: "custom-storage",
+	}
+	sc.VolumeBindingMode = &lateBinding
+
+	s := stack.NewStackBuilder("test-failure-pvc").
+		WithESMasterDataNodes(3, stack.DefaultResources).
+		WithPersistentVolumes("not-data", &sc.Name).
+		WithPersistentVolumes(volume.ElasticsearchDataVolumeName, &sc.Name) // create an additional volume that is not our data volume
+
+	var clusterUUID string
+	var seenPVCs []string
+	var killedPod corev1.Pod
+	var survivingPodNames []string
+
+	helpers.TestStepList{}.
+		WithSteps(stack.CreateStorageClass(*sc, k)).
+		WithSteps(stack.InitTestSteps(s, k)...).
+		WithSteps(stack.CreationTestSteps(s, k)...).
+		WithSteps(stack.RetrieveClusterUUIDStep(s.Elasticsearch, k, &clusterUUID)).
+		// Simulate a pod deletion
+		WithSteps(stack.PauseReconciliation(s.Elasticsearch, k)).
+		WithSteps(helpers.TestStepList{
+			{
+				Name: "Kill a node",
+				Test: func(t *testing.T) {
+					pods, err := k.GetPods(helpers.ESPodListOptions(s.Elasticsearch.Name))
+					require.NoError(t, err)
+					require.True(t, len(pods) > 0, "need at least one pod to kill")
+					for i, pod := range pods {
+						if i == 0 {
+							killedPod = pod
+						} else {
+							survivingPodNames = append(survivingPodNames, pod.Name)
+						}
+					}
+					err = k.DeletePod(killedPod)
+					require.NoError(t, err)
+				},
+			},
+			{
+				Name: "Wait for pod to be deleted",
+				Test: helpers.Eventually(func() error {
+					pod, err := k.GetPod(killedPod.Name)
+					if err != nil && !apierrors.IsNotFound(err) {
+						return err
+					}
+					if apierrors.IsNotFound(err) || killedPod.UID != pod.UID {
+						return nil
+					}
+					return fmt.Errorf("pod %s not deleted yet", killedPod.Name)
+				}),
+			},
+			{
+				Name: "Modify the es-data PVCs labels",
+				Test: helpers.Eventually(func() error {
+					pvcs, err := pvc.ListVolumeClaims(k.Client, s.Elasticsearch)
+					if err != nil {
+						return err
+					}
+					for _, pvc := range pvcs {
+						seenPVCs = append(seenPVCs, pvc.Name)
+						if pvc.Labels[label.VolumeNameLabelName] == volume.ElasticsearchDataVolumeName &&
+							pvc.Labels[label.PodNameLabelName] == killedPod.Name {
+							// this should ensure that when we resume reconciliation the operator creates a new PVC
+							// we also test correct reuse by keeping the non-data volume claim around and unchanged
+							delete(pvc.Labels, label.VolumeNameLabelName)
+							if err := k.Client.Update(&pvc); err != nil {
+								return err
+							}
+						}
+					}
+					return nil
+				}),
+			},
+			stack.ResumeReconciliation(s.Elasticsearch, k),
+			{
+				Name: "No PVC should have been reused for elasticsearch-data",
+				Test: helpers.Eventually(func() error {
+					pods, err := k.GetPods(helpers.ESPodListOptions(s.Elasticsearch.Name))
+					if err != nil {
+						return err
+					}
+					for _, pod := range pods {
+						if stringsutil.StringInSlice(pod.Name, survivingPodNames) {
+							continue
```

Comment:
At this point chances are the deleted pod is not back into the cluster yet, so we only iterate on pods we don't care about, continue on each one, then return nil? Which skips the test entirely?
Should we run `WithSteps(stack.CheckStackSteps(s, k)...). ` first, then this test? Also probably simpler to get the expected pod through its name directly instead of filtering on pods we don't care about?

Reply:
This comment was pure gold. You should get a🥇 for that. It surfaced that the actual fix was not fixing anymore since I merged master into this branch 😞 which was hidden by this flaw in the test ...